sasmodels/kerneldll.py

```diff
@@ -130,6 +130,9 @@
     # On Mac users will need the X code command line tools installed.
     #COMPILE = "gcc-mp-4.7 -shared -fPIC -std=c99 -fopenmp -O2 -Wall %s -o %s -lm -lgomp"
     CC = "cc -shared -fPIC -std=c99 -O2 -Wall".split()
+    CC.extend(os.environ.get("CPPFLAGS", "").split())
+    CC.extend(os.environ.get("CFLAGS", "").split())
+    CC.extend(os.environ.get("LDFLAGS", "").split())
```

Comment:
Should probably do this as (untested):
```
import shlex
...
CC = os.environ.get("CC", "cc")
CFLAGS = os.environ.get("CFLAGS", "-std=c99 -O2 -Wall -fpic")
LDFLAGS = os.environ.get("LDFLAGS", "-shared")
CC = shlex.split(" ".join((CC, CFLAGS, LDFLAGS)))
LIBS = shlex.split(os.environ.get("LIBS", "-lm"))
...
    return CC + [source, "-o", output] + LIBS
```

`shlex.split()` claims to handle quoted arguments properly.  For example,
```
    CC='"/Users/joe/gcc 9.2/build/bin/cc"'
```

Need to document the default values for these variables so that the user who wants to override them can choose to include them or not.

This still leaves the problem that the compile command uses "-o outfile" to specify the file, so for example you wouldn't have enough control to build a MSVC style `cl` command (not that you would need to do so on linux or mac).  I don't know if the alternative unix compilers such as pgcc or icc happen to use the same "-o" flag as the default `cc`.  It should since this is part of the [posix standard cc](https://pubs.opengroup.org/onlinepubs/7908799/xcu/cc.html).